core/src/main/java/org/apache/druid/utils/ConnectionUriUtils.java

```diff
@@ -57,6 +67,194 @@ public static void throwIfPropertiesAreNotAllowed(
     }
   }
 
+  public static Set<String> tryParseJdbcUriParameters(String connectionUri, boolean allowUnknown)
+  {
+    if (connectionUri.startsWith(MYSQL_PREFIX)) {
+      try {
+        return tryParseMySqlConnectionUri(connectionUri);
+      }
+      catch (ClassNotFoundException notFoundMysql) {
```

Comment:
are these branches covered by unit tests?

Reply:
They are not currently, trying to think of a good way to test it, i guess i need some different classloaders to model different configurations or something. Any ideas?

Reply:
one way would be to take `Class.forName()` to a different class and use that here instead. In tests, you could mock the other class and method.

Reply:
I found `Mockito.mockStatic` which lets me model these branches a bit by just letting me mock the `tryParse{x]ConnectionUri` methods, so the coverage is a bit better now i think